Kuadrant/multicluster-gateway-controller
# #98: Add DNSPolicy API and bare controller

## Description

> Closes #98 

Add CRD and empty controller for DNSPolicy resource, including spec for health checks.

## Verification steps

1. Run local setup
    ```sh
    make local-setup
    ```

1. Verify that the CRD is created in the control cluster
    ```sh
    kubectl get crd dnspolicies.kuadrant.io                                                                                                                                      
    NAME                      CREATED AT
    dnspolicies.kuadrant.io   2023-03-29T10:29:11Z
    ```

1. Start the controller
    ```sh
    (export $(cat ./controller-config.env | xargs) && export $(cat ./aws-credentials.env | xargs) && make build-controller install run-controller
    ```

1. Verify that the DNSPolicy controller is started
    ```sh
    2023-03-29T11:57:22+01:00       INFO    Starting EventSource    {"controller": "dnspolicy", "controllerGroup": "kuadrant.io", "controllerKind": "DNSPolicy", "source": "kind source: *v1alpha1.DNSPolicy"}                                                                                                                                                                
    2023-03-29T11:57:22+01:00       INFO    Starting Controller     {"controller": "dnspolicy", "controllerGroup": "kuadrant.io", "controllerKind": "DNSPolicy"} 
    ```
